core/src/main/java/org/radargun/stages/cache/background/PrivateLogLogic.java

```diff
@@ -48,12 +51,13 @@ protected boolean invokeLogic(long keyId) throws Exception {
          if (prevValue == null || !prevValue.contains(prevOperation.operationId)) {
             // non-cleaned old value or stale read, try backup
             backupValue = checkedGetValue(~keyId);
-            if (backupValue == null || !backupValue.contains(prevOperation.operationId)) {
+            // Modifying the same key within a single transaction may cause false stale reads, avoid it by checking txModificationKeyIds
```

Comment:
I would reconsider other options to get around the stale reads, since having multiple modifications of the same entry within single transaction can reveal bugs, too.